app/helpers/application_helper/button/catalog_item_button.rb

```diff
@@ -0,0 +1,9 @@
+class ApplicationHelper::Button::CatalogItemButton < ApplicationHelper::Button::Basic
+  delegate :role_allows_any?, :to => :current_user
```

Comment:
so the 2 delefate lines are needed because you call `role_allows_any?` once?

I'd say that you can just call `@view_context.current_user.role_allows_any?(....)`

If It's a single occurrence.the it does not justify the 2 delegations.